hcl2template/addrs/plugin.go

```diff
@@ -218,7 +218,20 @@ func ParsePluginSourceString(str string) (*Plugin, hcl.Diagnostics) {
 		})
 	}
 
-	return &Plugin{
+	plug := &Plugin{
 		Source: str,
-	}, diags
+	}
+	if len(plug.Parts()) > 16 {
+		return nil, diags.Append(&hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  "Too many parts to source URL",
+			Detail: fmt.Sprintf(`The source URL %q must have at most 16 components, and the one provided has %d.`+
+				`This is unsupported by Packer, please consider using a source that has less components to it, `+
+				`or if this is a blocking issue for you, please open an issue to ask for supporting more `+
+				`components to the source URI.`,
+				str, len(plug.Parts())),
+		})
+	}
+
+	return plug, diags
```

Comment:
I would expect this to be nil and not the plugin being instantiated above. 
What is the state of plug at this point? 

```
     // Maybe reorder things so that you can return without plug    
	return nil, diags
```

nit: The error message should be broken into separate lines 
```
~>  pakcer plugins install --path packer-plugin-hashicups_v1.0.2-dev_x5.0_darwin_arm64 github.com/hashicorp/azr/a/b/c/d/e/f/g/h/i/j/k/l/m/hashicups
<nil>: Too many parts to source URL; The source URL "github.com/hashicorp/azr/a/b/c/d/e/f/g/h/i/j/k/l/m/hashicups" must have at most 16 components, and the one provided has 17.This is unsupported by Packer, please consider using a source that has less components to it, or if this is a blocking issue for you, please open an issue to ask for supporting more components to the source URI.
```

Reply:
Oh, I would have expected the diag to be formatted for the output, I'll fix that!